src/test/java/org/openmrs/contrib/qaframework/page/AttachmentsPage.java

```diff
@@ -30,9 +37,20 @@ public AttachmentsPage(Page page) {
 		super(page);
 	}
 
-	public void setFileUrl(String fileUrl) {
+	public String createRandomFile() throws IOException {
+		File tempFile = File.createTempFile("file", ".pdf");
+		tempFile.deleteOnExit();
+		try (InputStream pdf = getClass().getClassLoader().getResourceAsStream("data/file.pdf")) {
+			try (Writer w = new FileWriter(tempFile)) {
+				IOUtils.copy(pdf, w, StandardCharsets.UTF_8);
+			}
+		}
+		return tempFile.getAbsolutePath();
+	}
+	
+	public void setFileUrl()throws IOException {
 		WebElement element = driver.findElement(By.xpath("//input[@type='file']"));
-		element.sendKeys(fileUrl);
+		element.sendKeys(createRandomFile());
 	}
 
```

Comment:
@kdaud does the above method setFileUrl() look better? I have refactored it to handle the file path returned by the createRandomFile() right here in the attachmentsPage. Let me know if it makes sense or if it doesn't.

Reply:
@Irenyak1 if the refactoring of the method makes the function when invoked to handle two things(can pass a url to hard coded path and passing a createRandonFile()) then that will be a good approach. But if refactoring the method gets rid of handling the handling hard coded path then its better not to refactor it but rather create another method to handle createRandonFile().

Reply:
@kdaud it does not handle the hard coded. Then I think I should just create one.